clients/ordering/src/components/card.tsx

```diff
@@ -0,0 +1,160 @@
+import Collapse from '@material-ui/core/Collapse';
+import Avatar from '@material-ui/core/Avatar';
+import IconButton from '@material-ui/core/IconButton';
+import Typography from '@material-ui/core/Typography';
+import red from '@material-ui/core/colors/red';
+import FavoriteIcon from '@material-ui/icons/Favorite';
+import ShareIcon from '@material-ui/icons/Share';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import AddBoxIcon from '@material-ui/icons/AddBox';
+import Grid from '@material-ui/core/Grid';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    card: {
+      maxWidth: 400,
+      width: 400,
+      height: 435,
+    },
+    media: {
+      height: 0,
+      paddingTop: '56.25%', // 16:9
+      marginTop: '30',
+    },
+    decription: {
+      height: 50,
+    },
+    actions: {
+      display: 'flex',
+      alignItems: 'end',
+    },
+    expand: {
+      transform: 'rotate(0deg)',
+      marginLeft: 'auto',
+      transition: theme.transitions.create('transform', {
+        duration: theme.transitions.duration.shortest,
+      }),
+    },
+    expandOpen: {
+      transform: 'rotate(180deg)',
+    },
+    avatar: {
+      backgroundColor: red[500],
+    },
+  });
+
+type CardsProps = WithStyles<typeof styles>;
+
+export interface MenuProps extends WithStyles<typeof styles> {
+  avatar: string;
+  title: string;
+  subheader: string;
+  image: string;
+  description: string;
+}
+
+const MenuCard: React.FC<MenuProps> = ({
+  avatar,
+  title,
+  subheader,
+  image,
+  description,
+  classes,
+}) => {
+  const [expanded, setExpanded] = useState<boolean>(false);
+
+  return (
+    <Grid item={true}>
+      <Card className={classes.card}>
+        <CardHeader
+          avatar={
+            <Avatar aria-label="Item" className={classes.avatar}>
+              {avatar}
+            </Avatar>
+          }
+          action={
+            <IconButton>
+              <MoreVertIcon />
+            </IconButton>
+          }
+          title={title}
+          subheader={subheader}
+        />
+        <CardMedia
+          className={classes.media}
+          image={image}
+          title={title}
+          style={{ height: 0, paddingTop: '56.25%' }}
+        />
+        <CardContent className={classes.decription}>
+          <Typography component="p">{description}</Typography>
+        </CardContent>
+        <CardActions className={classes.actions} disableActionSpacing={true}>
+          <IconButton aria-label="Add to Cart">
+            <AddBoxIcon />
+          </IconButton>
+          <IconButton aria-label="Add to Favorites">
+            <FavoriteIcon />
+          </IconButton>
+          <IconButton aria-label="Share">
+            <ShareIcon />
+          </IconButton>
+          <IconButton
+            className={classes.expand}
+            aria-label="Show more"
+            onClick={() => setExpanded(!expanded)}
```

Comment:
```suggestion
            onClick={() => setExpanded(prevExpanded => !prevExpanded)}
```
This will likely do what you want, but an updater function is safer to use when updating the state by using the previous state value. React may batch several updates into one update, so this ensures that the state update is dependent on the true value of the previous state